### PR TITLE
mitx combined program requirements

### PR DESCRIPTION
**Pull request by abeglova**
closes https://github.com/mitodl/ol-data-platform/issues/663
closes https://github.com/mitodl/ol-data-platform/issues/664
closes https://github.com/mitodl/ol-data-platform/issues/665

This pr adds the MITx combined programs, courses and program requirements tables

### CODE REVIEW

File: src/ol_dbt/models/intermediate/mitx/int__mitx__courses.sql [hunk 1]
```diff
@@ -0,0 +1,44 @@
+with mitxonline_courses as (
+    select
+        course_id as mitxonline_course_id
+        , course_title
+        , course_is_live
+        , course_readable_id
+        , trim(course_number) as course_number
```
Comment (rachellougee) [inline src/ol_dbt/models/intermediate/mitx/int__mitx__courses.sql:7]: Is there white spacing in course_number? should we trim it in staging?

File: src/ol_dbt/models/intermediate/mitxonline/int__mitxonline__program_requirements.sql
Comment (rachellougee) [inline src/ol_dbt/models/intermediate/mitxonline/int__mitxonline__program_requirements.sql:1]: The logic is complicated. These two nested courses are a bit confusing and appear twice as electives with different required numbers, how are these two courses supposed to query?

File: src/ol_dbt/models/intermediate/mitx/int__mitx__courses.sql (hunk 1 above)
Comment (abeglova) [reply src/ol_dbt/models/intermediate/mitx/int__mitx__courses.sql:7]: Actually I did, i just forgot to remove the trim here

File: src/ol_dbt/models/intermediate/mitxonline/int__mitxonline__program_requirements.sql [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        , operator_nodes.programrequirement_title
+        , operator_nodes.programrequirement_path
+        , operator_nodes.programrequirement_operator_value
+            as electiveset_required_number
+        , case operator_nodes.programrequirement_operator
+            when 'min_number_of' then 'Elective'
+            when 'all_of' then 'Core' end
+            as programrequirement_type
+
+        , ROW_NUMBER()
```
Comment (rachellougee) [inline src/ol_dbt/models/intermediate/mitxonline/int__mitxonline__program_requirements.sql:33]: Can you fix these uppercase functions from this file to be consistent with the rest of the code?

**Review by rachellougee — APPROVED** [review]
LGTM

**Pull request merged**